src/extract.rs

```diff
@@ -3,15 +3,15 @@ use futures::prelude::*;
 use crate::{configuration::Store, Request, Response, RouteMatch};
 
 /// An extractor for an app with `Data`
-pub trait Extract<Data>: Send + Sized + 'static {
+pub trait Extract<Param, Data>: Send + Sync + Sized + 'static where Param: Send + Sized + 'static {
```

Comment:
I feel like the trait signature should be `Extract<Seed, Data>`. This looks quite reversed; for instance, `Json` is a extractor, not `()`.

Reply:
`()` is intended as an extractor for `Json`.  That sounds more plausible even, `()` extracts `Param` from `Data`. If we go the way of adding a trait rather than replacing one, this could of course be solved with another name. I'm open to bikeshedding.

However, `Extract<(), Data> for Json` would mean that the natural `&self` argument becomes `&Seed`.